drivers/gpio/gpio_pca95xx.c

```diff
@@ -455,9 +602,14 @@ static const struct gpio_driver_api gpio_pca95xx_drv_api_funcs = {
 	.port_set_bits_raw = gpio_pca95xx_port_set_bits_raw,
 	.port_clear_bits_raw = gpio_pca95xx_port_clear_bits_raw,
 	.port_toggle_bits = gpio_pca95xx_port_toggle_bits,
+#ifdef CONFIG_PCA95XX_INTERRUPT_SUPPORT
 	.pin_interrupt_configure = gpio_pca95xx_pin_interrupt_configure,
```

Comment:
The syscall wrappers require that this field be assigned to the API regardless of whether interrupts are supported.  Put the conditional inside the implementation of the function.